### PR TITLE
fix improper use of current-chat-id and rename subs

### LINKED ISSUE #18391: Group messages are automatically getting marked read
### Steps
1. Open group details page
2. Receive new messages while **staying** on group details page.

https://github.com/status-im/status-mobile/assets/17097240/1343dfef-fb4f-4ef7-b7bb-01c1b5fcce2e

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-mobile/issues/18391

### Summary

- Removes use `:current-chat-id` just for checking group details
- Removes use of anonymous function in functional component
- Fixes names of subscriptions

### Testing
Please make sure management of group is working fine (removing, adding of members etc.)

status: ready

### CODE REVIEW

File: src/status_im/contexts/chat/group_details/view.cljs [hunk 1]
```diff
@@ -151,84 +149,78 @@
                         :on-press show-profile-actions}})
      item]))
 
-(defn f-group-details
-  []
-  (fn []
-    (rn/use-effect (fn []
-                     #(rf/dispatch [:chat/close])))
```
Comment (ibrkhalil) [inline src/status_im/contexts/chat/group_details/view.cljs:158]: Same here

File: src/status_im/contexts/chat/group_details/view.cljs [hunk 2]
```diff
@@ -20,9 +20,7 @@
     :icon-only?          true
     :container-style     {:margin-left 20}
     :accessibility-label :back-button
-    :on-press            (fn []
-                           (rf/dispatch [:navigate-back])
-                           (rf/dispatch [:chat/close]))}
+    :on-press            #(rf/dispatch [:navigate-back])}
```
Comment (ibrkhalil) [inline src/status_im/contexts/chat/group_details/view.cljs:23]: We need to close currently active group chat or it'll keep `current-chat-id` in app-db causing unread messages to be set as read immediately (No unread indicator will be shown).

Comment (ibrkhalil) [inline src/status_im/contexts/chat/group_details/view.cljs:23]: We need to close currently active group chat or it'll keep `current-chat-id` in app-db causing unread messages to be set as read immediately (No unread indicator will be shown).

File: src/status_im/contexts/chat/group_details/view.cljs (hunk 1 above)
Comment (ibrkhalil) [inline src/status_im/contexts/chat/group_details/view.cljs:158]: Same here

File: src/status_im/contexts/chat/group_details/view.cljs (hunk 2 above)
Comment (ibrkhalil) [reply src/status_im/contexts/chat/group_details/view.cljs:23]: Check [here](https://github.com/status-im/status-mobile/pull/18149)

Comment (Parveshdhull) [reply src/status_im/contexts/chat/group_details/view.cljs:23]: hi @ibrkhalil,
Thank you very much for reviewing PR.
This is what this PR is fixing. Its is removing opening of chat, just to check group details. (fix improper use of current-chat-id)

Comment (ibrkhalil) [reply src/status_im/contexts/chat/group_details/view.cljs:23]: I see, We no longer add current-chat-id when opening group details.
Thanks for the explanation Parvesh!

**Review by ibrkhalil — APPROVED** [review]

**Comment by pavloburykh** [comment]
@Parveshdhull thanks for the PR. Failed e2e are known and not related to the PR.

**Comment by Parveshdhull** [comment]
hi @pavloburykh,

Thank you very much for checking the e2e tests. But it looks currently adding-removing members from home screen is not working because not only group details screen but most of the events related to management of group are also relying on current-chat-id.

PR will also going to need manual-qa-testing as bug is more complicated then it initially looked. Will ping qa once PR is ready to test.

**Comment by Parveshdhull** [comment]
No new changes, just rebased PR.

**Comment by qoqobolo** [comment]
@Parveshdhull thanks for PR!
I see three issues with the group chat members management atm, one of them is only in PR and the other two in nightly. 

> But it looks currently adding-removing members from home screen is not working because not only group details screen but most of the events related to management of group are also relying on current-chat-id.

By this, did you mean that in this PR we need to check that everything works the same as in nightly, or that you also fix other issues related to the group management (in this case, we'll log the last two separately)?

### ISSUE 1 (PR only): Group members list is not displayed in group details anymore

**PR**
Screenshot 2024-01-10 at 17 27 32

**Nightly**
Screenshot 2024-01-10 at 17 27 57

---

### ISSUE 2 (nightly and PR): Non-admin can "remove" a group member

**Steps:**
1. UserA adds UserB and UserC to contacts and creates a group
2. UserC removes UserA or UserB from the group

In this case, the UI behaves weirdly: the system message `UserC has added` appears, and UserA is not removed from the group in fact. 

Screenshot 2024-01-10 at 17 57 57

https://github.com/status-im/status-mobile/assets/67952253/16a8ff7e-5572-42aa-b8e9-4164dc7a92ef



As I can see from [Figma](https://www.figma.com/file/7KIYbhoqNGAIFonE0w9TDz/Messages-for-Mobile?type=design&node-id=2105-522742&mode=design&t=MAO0lXxmaOXElzuh-4) a non-admin should not have this ability to remove user, but I might be wrong. Do you know for sure? If not, we can check this with the designers.

### ISSUE 3 (nightly and PR): Name of a new member is not displayed in the system message for other users if they are not contacts / before receiving first message from the new member

**Steps:**
1. UserA adds UserB and UserC to contacts and creates a group
2. UserC adds UserD to contacts and adds them to the group

This is how looks for UserA and UserB:
Screenshot 2024-01-10 at 18 06 34

**Comment by Parveshdhull** [comment]
Hi @qoqobolo, Thank you very much for testing the PR and finding these issues.

>  did you mean that in this PR we need to check that everything works the same as in nightly, or that you also fix other issues related to the group management (in this case, we'll log the last two separately)?

Sorry for the confusion, I meant please check everything works the same as in nightly + fixes the linked bug https://github.com/status-im/status-mobile/issues/18391

So, yes please log last two issues separately. Issue 1 should be fixed now.

**Comment by qoqobolo** [comment]
Thanks for the fixes @Parveshdhull!
Don't see any other issues, PR can be merged.
Nightly issues will be logged separately.

**Pull request merged**